opennms-webapp/src/main/java/org/opennms/web/controller/alarm/ThresholdBoxController.java

```diff
@@ -0,0 +1,135 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2007-2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2014 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.controller.alarm;
+
+import org.opennms.netmgt.dao.api.AlarmRepository;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.web.alarm.AcknowledgeType;
+import org.opennms.web.alarm.AlarmUtil;
+import org.opennms.web.alarm.SortStyle;
+import org.opennms.web.alarm.filter.AlarmCriteria;
+import org.opennms.web.alarm.filter.PartialUEIFilter;
+import org.opennms.web.filter.Filter;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.Assert;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.mvc.AbstractController;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * A controller that handles querying threshold alarms from the alarm table to create the front-page
+ * threshold alarm box.
+ *
+ * @author <a href="mailto:ronny@opennms.org">Ronny Trommer</a>
+ */
+public class ThresholdBoxController extends AbstractController implements InitializingBean {
+    public static final int DEFAULT_ROWS = 16;
+
+    private static final String UEI_DEFAULT_FILTER = "uei.opennms.org/threshold/";
+    private static final String UEI_FILTER_PROPERTY = "opennms.thresholdAlarm.uei.filter";
+    private static final String THRESHOLD_ALARM_COUNT_PROPERTY = "opennms.thresholdAlarm.count";
+
+    private AlarmRepository m_webAlarmRepository;
+    private String m_successView;
+    private AlarmCriteria m_alarmCriteria;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected ModelAndView handleRequestInternal(HttpServletRequest request, HttpServletResponse response) throws Exception {
+
+        int rows = DEFAULT_ROWS;
+
+        if (System.getProperty(THRESHOLD_ALARM_COUNT_PROPERTY) != null) {
```

Comment:
Aren't the Boolean.getBoolean(), Integer.getInteger() and Long.getLong() methods considered a notorious landmine in the Java API? Sonar probably has a rule that flags any usage of them. :)

http://konigsberg.blogspot.com/2008/04/integergetinteger-are-you-kidding-me.html

I like the Boolean.getBoolean("true") == Boolean.FALSE example.